Src/Test.UnitTests.RE2.Managed/Regex2Tests.cs

```diff
@@ -190,18 +190,21 @@ public void Regex2_MatchWithNamedGroups()
             Dictionary<string, int> groupName2Index;
             Dictionary<int, string> index2GroupName;
             List<string> submatchStrings;
+            List<MatchGroup> matchGroups;
 
-            isMatch = Regex2.Matches(@"abc", "def", out _, out _, out _);
+            isMatch = Regex2.Matches(@"abc", "def", out _, out _, out _, out _);
             Assert.False(isMatch);
 
-            isMatch = Regex2.Matches(@"abc", "abc", out groupName2Index, out index2GroupName, out submatchStrings);
+            isMatch = Regex2.Matches(@"abc", "abc", out groupName2Index, out index2GroupName, out submatchStrings, out matchGroups);
             Assert.True(isMatch);
             Assert.Empty(groupName2Index);
             Assert.Empty(index2GroupName);
             Assert.Single(submatchStrings);
             Assert.Equal("abc", submatchStrings[0]);
+            Assert.Single(matchGroups);
+            Assert.Equal(new MatchGroup(MatchGroupType.Full, "abc"), matchGroups[0]);
 
-            isMatch = Regex2.Matches(@"(?P<g1>a)(b)(?P<g2>c)", "abc", out groupName2Index, out index2GroupName, out submatchStrings);
+            isMatch = Regex2.Matches(@"(?P<g1>a)(b)(?P<g2>c)", "abc", out groupName2Index, out index2GroupName, out submatchStrings, out matchGroups);
```

Comment:
>abc [](http://example.com/codeflow?start=64&length=3)

can you make the following test:
use "abc abc abc"

this should show 3 matches, and in each match we should see each group.

Check this output:
var matches = Regex.Matches("abc abc abc", @"(?<1>a)(b)(?<2>c)");

Reply:
I've updated the test. Are these the results you are expecting?